src/manager.rs

```diff
@@ -48,14 +48,16 @@ pub struct ActorManager<A: Actor> {
     pub(crate) ctx: Context<A>,
 }
 
-impl<A: Actor> ActorManager<A> {
+impl<A: Actor<Stop = ()>> ActorManager<A> {
```

Comment:
Should this perhaps not accept any stop type?

Reply:
I've split it out because we would have to do:

```rust
spawner.spawn(async move {
	let _ = fut.await;
});
```

I am not sure if it is good to drop the returned value here?

Reply:
Hm. I think there's something to be said for having a function that takes an actor and returns an address - I think that's probably a fairly common usecase. Thus far anyway, in all the code I've written I haven't needed to return anything from `stopped`, so I think that goes to show that there are usecases where you don't care about the return value. If you *do* care about it and still want to `spawn`, one could do:

```rust
let (addr, fut) = actor.create(/* cap */).run();
let ret = tokio::spawn(fut);
```

Otherwise, there could be yet another method like
```rust
let (addr, ret) = actor.create(/* cap */).foobar();
```

which does this for you. I'm not sure if this is needed though? This would require some internal oneshot channel type thing to abstract over all the different task abstractions. It'd also be different to a `smol` `Task` for instance, since dropping it would not cancel it, which might be weird. Thoughts?

Reply:
> I think there's something to be said for having a function that takes an actor and returns an address

Definitely. And as long as `Stop` is set to `()`, nothing changes with this patch. You can still use `.spawn()`.

I don't think it is worth creating another method for spawning and returning. For the uncommon case where `Stop` returns something other than `()`, the additional `.run()` is barely worth mentioning the complexity IMO.

On the other hand, if we make this function take actors with any `Stop` value, we are removing an ability for the compiler to guide the user towards what is likely the correct behaviour: Caring about the return value if it is `!= ()`.

Reply:
> I think there's something to be said for having a function that takes an actor and returns an address

Definitely. And as long as `Stop` is set to `()`, nothing changes with this patch. You can still use `.spawn()`.

I don't think it is worth creating another method for spawning and returning. For the uncommon case where `Stop` returns something other than `()`, the additional `.run()` is barely worth mentioning the complexity IMO.

On the other hand, if we make this function take actors with any `Stop` value, we are removing an ability for the compiler to guide the user towards what is likely the correct behaviour: Caring about the return value if it is `!= ()`.

Reply:
> On the other hand, if we make this function take actors with any Stop value, we are removing an ability for the compiler to guide the user towards what is likely the correct behaviour: Caring about the return value if it is != ().

That's true. We can keep it this way then I think.